README.md

```diff
@@ -41,104 +41,123 @@ firestoreService.INITIALIZE(firebaseConfig);
 
 ```
 
-## HTTP Methods
+## Response format
 
-For all the examples we will be using the following database.
-
-### RESPONSE
-
-All the http methods will return a response with the following body.
+Every HTTP method will return a response with the following body.
 
 ```
 {
   ok: a boolean -> true: Success, false: Failure
   data: The data you requested,
-  status: An http Code see the code table below,
+  status: An HTTP status code, see the code table below,
   statusText: 'OK' or 'Failure',
   request: the actual request (GET, POST, etc.)
 }
 ```
 
+## Supported methods
+
 ### GET
 
-You can get all the elements from a collection by using get with the path towards the collection.
+You can get all the elements from a collection or a specific element if the element's id is specified.
 
-Note: Have in mind that the path will be 'collection/id/collection2/id2/.../collection'
-It will always finish with a collection.
+Note: The path will always be `collection/id/collection2/id2/...`. The "url" will finish with a collection if you want an array of elements or with an id if you want a single one.
 
-```js
-const path = 'regions'
+Eg:
 
-const response = await firestoreService.GET(path);
+```js
+const response = await firestoreService.GET('regions');
 ```
 
-In the response there will be all the regions
+The response will have all the "regions"
 
 ```js
-const path = 'regions?id=NA'
-
-const response = await firestoreService.GET(path);
+const response = await firestoreService.GET('regions/32');
 ```
 
-In the response there will be all the attributes of the NA document.
+The response will have the information about the region with id 32.
 
 
 ### POST
 
-You can update an element by calling POST with the path towards the collection, the id of the item to update and the body.
+You can create an element by calling POST with the collection's path and the body with the element's data.
```

Comment:
Do an upsert of

Reply:
but it's not an update, it's just a creation. Not sure I'm getting this right.

Reply:
If the document is inside the collection it'll do an update, if not, will create a new one

Reply:
the document shouldn't be edited if it's a post. If a document with the same data is posted the document is created with another id than the original so it will be duplicated (except for the id), in case there's a restriction in one field that should be unique like the username and someone tries to do a post with a duplicated username the original one shouldn't be edited, the response should be an error with a 400 status code.

Reply:
If it currently our implementation does what you are saying we should change the implementation since that's not how an api should work.

Reply:
I agree

Reply:
Actually right now with the last refactor of Post and Put, when you want to create an element with POST you don't have the id. So you are always creating a new one and the response will be the id of the created element. If you want to modify an element you use PUT or PATCH with the id of the element you want to modify. @FrankIglesias @SKOLZ

Reply:
@LucasZibell can we return the whole element in the response? instead of just the id?